palantirnet/federated-search-demo
# PL-52: Address Domain Mismatches

## JIRA Ticket(s)

- See ticket [PL-52: Drupal 8 demo content links / site names are incorrect.](https://palantir.atlassian.net/browse/PL-52).

## Description

The demo site was not listing results correctly because two Drupal sites (single / domain access) were using the same configuration and because some of the domains in configuration were incorrect.

- Use the `fs-demo` domain for all sites and use different subdomains.
- Create a config split for the Domain Access solr index so that the site name can be different. See `field_settings.site_name.configuration.site_name`.
- Create a config split for the Domain Access solr settings so that the hash can be different. Otherwise the document keys are the same and overwrite each other. See `site_hash`.
- Update the build order to allow config overrides for Domain Access to take into affect and avoids overwriting of Solr documents because of identical IDs.

## Testing instructions

1. Install the site per the normal direction, `phing build install-all`.
2. Go to a sample search page and search for a Drupal 8 term, `pasta`, http://d8.fs-demo.local/search-app?search=pasta. Confirm that an entry shows up for the single site and for the domain access site. Confirm that the images and links work.
3. Search for a D7 term, `ex`, http://d8.fs-demo.local/search-app?search=ex. Confirm that the entry shows up, that the URL / Image shows up and try the link.

## Relevant Screenshots/gifs:

![example](https://user-images.githubusercontent.com/397902/50611979-fe437980-0e9d-11e9-8a43-08d2e04ad0f3.jpg)


---

[Pull Request Guidelines](https://github.com/palantirnet/development_documentation/blob/master/guidelines/pr_review.md)
